src/wallet.rs

```diff
@@ -364,6 +376,94 @@ pub fn process_command(config: &WalletConfig) -> Result<String, Box<error::Error
                 ))?,
             }
         }
+        // Deploy a custom program to the chain
+        WalletCommand::Deploy(ref program_location) => {
+            let params = json!(format!("{}", config.id.pubkey()));
+            let balance = RpcRequest::GetBalance
+                .make_rpc_request(&config.rpc_addr, 1, Some(params))?
+                .as_i64();
+            if let Some(tokens) = balance {
+                if tokens < 1 {
+                    Err(WalletError::DynamicProgramError(
+                        "Insufficient funds".to_string(),
+                    ))?
+                }
+            }
+
+            let last_id = get_last_id(&config)?;
+            let program = Keypair::new();
+            let program_userdata = elf::File::open_path(program_location)
+                .map_err(|_| {
+                    WalletError::DynamicProgramError("Could not parse program file".to_string())
+                })?.get_section(PLATFORM_SECTION_C)
+                .ok_or_else(|| {
+                    WalletError::DynamicProgramError(
+                        "Could not find entrypoint in program file".to_string(),
+                    )
+                })?.data
+                .clone();
+
+            let tx = Transaction::system_create(
+                &config.id,
+                program.pubkey(),
+                last_id,
+                1,
+                program_userdata.len() as u64,
+                bpf_loader::id(),
+                0,
+            );
+            let signature_str = serialize_and_send_tx(&config, &tx)?;
+            if !confirm_tx(&config, &signature_str)? {
+                Err(WalletError::DynamicProgramError(
+                    "Program allocate space failed".to_string(),
+                ))?
+            }
+
+            let mut offset = 0;
+            let write_result = program_userdata
+                .chunks(USERDATA_CHUNK_SIZE)
+                .map(|chunk| {
+                    let tx = Transaction::write(
+                        &program,
+                        bpf_loader::id(),
+                        offset,
+                        chunk.to_vec(),
+                        last_id,
+                        0,
+                    );
+                    let signature_str = serialize_and_send_tx(&config, &tx).unwrap();
```

Comment:
By firing off all the `write()` transactions in parallel, we risk hitting AccountInUse errors since they all want to write to the same account and we now have an mt bank.

I ran into this in the SDK today, and added some retry logic: https://github.com/solana-labs/solana-web3.js/commit/d8a572420be1e621f802853b91dc54919c9282f6

We've probably just gotten lucky here so far.  I'm ok with this PR landing without retry logic added but if so please file a follow-up issue since I'm pretty sure we'll run into this when somebody tries to load a big program